--- gps-python2.py	(original)
+++ gps-python2.py	(refactored)
@@ -22,7 +22,7 @@
   def run(self):
     global gpsd
     while gpsp.running:
-      gpsd.next() #this will continue to loop and grab EACH set of gpsd info to clear the buffer
+      next(gpsd) #this will continue to loop and grab EACH set of gpsd info to clear the buffer
 
 if __name__ == '__main__':
   gpsp = GpsPoller() # create the thread
@@ -34,29 +34,29 @@
 
       os.system('clear')
 
-      print
-      print ' GPS reading'
-      print '----------------------------------------'
-      print 'latitude    ' , gpsd.fix.latitude
-      print 'longitude   ' , gpsd.fix.longitude
-      print 'time utc    ' , gpsd.utc,' + ', gpsd.fix.time
-      print 'altitude (m)' , gpsd.fix.altitude
-      print 'eps         ' , gpsd.fix.eps
-      print 'epx         ' , gpsd.fix.epx
-      print 'epv         ' , gpsd.fix.epv
-      print 'ept         ' , gpsd.fix.ept
-      print 'speed (m/s) ' , gpsd.fix.speed
-      print 'climb       ' , gpsd.fix.climb
-      print 'track       ' , gpsd.fix.track
-      print 'mode        ' , gpsd.fix.mode
-      print
-      print 'sats        ' , gpsd.satellites
+      print()
+      print(' GPS reading')
+      print('----------------------------------------')
+      print('latitude    ' , gpsd.fix.latitude)
+      print('longitude   ' , gpsd.fix.longitude)
+      print('time utc    ' , gpsd.utc,' + ', gpsd.fix.time)
+      print('altitude (m)' , gpsd.fix.altitude)
+      print('eps         ' , gpsd.fix.eps)
+      print('epx         ' , gpsd.fix.epx)
+      print('epv         ' , gpsd.fix.epv)
+      print('ept         ' , gpsd.fix.ept)
+      print('speed (m/s) ' , gpsd.fix.speed)
+      print('climb       ' , gpsd.fix.climb)
+      print('track       ' , gpsd.fix.track)
+      print('mode        ' , gpsd.fix.mode)
+      print()
+      print('sats        ' , gpsd.satellites)
 
       time.sleep(5) #set to whatever
 
   except (KeyboardInterrupt, SystemExit): #when you press ctrl+c
-    print "\nKilling Thread..."
+    print("\nKilling Thread...")
     gpsp.running = False
     gpsp.join() # wait for the thread to finish what it's doing
-  print "Done.\nExiting."
+  print("Done.\nExiting.")
 
